image.go

```diff
@@ -64,6 +64,13 @@ func (img DockerImage) DefaultLatestVersion() DockerImage {
 	return img
 }
 
+// Validate checks that the given docker image is valid.
+// Returns nil if valid, or an error if not valid.
+func (img DockerImage) Validate() error {
```

Comment:
This is needed to validate a docker image after a field of the struct has been changed.
See giantswarm/app-service#306